common/src/main/java/tachyon/resource/ResourcePool.java

```diff
@@ -66,19 +67,33 @@ protected ResourcePool(int maxCapacity, BlockingQueue<T> resources) {
    */
   public T acquire() {
     // If the resource pool is empty but capacity is not yet full, create a new resource.
-    synchronized (mCapacityLock) {
-      if (mResources.isEmpty() && mCurrentCapacity < mMaxCapacity) {
-        T newResource = createNewResource();
-        mCurrentCapacity ++;
-        return newResource;
-      }
+    T resource = mResources.poll();
```

Comment:
The comment above this is no longer accurate. I think this comment can be moved down to the following block.

Also, add a new comment about this block of code.